narwhals/_pandas_like/series.py

```diff
@@ -227,11 +227,8 @@ def ewm_mean(
 
     def scatter(self: Self, indices: int | Sequence[int], values: Any) -> Self:
         if isinstance(values, self.__class__):
-            # .copy() is necessary in some pre-2.2 versions of pandas to avoid
-            # `values` also getting modified (!)
-            _, values = align_and_extract_native(self, values)
             values = set_index(
-                values.copy(),
+                values._native_series,
```

Comment:
the extra copy may no longer be necessary as of https://github.com/narwhals-dev/narwhals/pull/1159 ?